src/reqstool/requirements_indata/requirements_indata_paths.py

```diff
@@ -40,12 +41,23 @@ def prepend_paths(self, prepend_dir):
         for prop in dir(self):
             if (
                 not prop.startswith("__")
-                and isinstance(getattr(self, prop), RequirementsIndataPathItem)
+                and self.attribute_is_instance_or_list_of_class(prop_name=prop, the_class=RequirementsIndataPathItem)
                 and prop not in ["requirements_yml", "svcs_yml", "mvrs_yml"]
             ):
                 path_item = getattr(self, prop)
-                if path_item.path is not None:
-                    path_item.path = str(PurePath(prepend_dir, path_item.path))
+                self.check_type_and_prepend_path(path_item=path_item, prepend_dir=prepend_dir)
+
+    # prepend path for each entry in a list of RequirementsIndataPathItem(s) or just on a single entity of the class
```

Comment:
if there is a need to comment the code then document according to our internal development manual

Reply:
fixed!